[MSSQL] Fix file exist error

Dfte opened the pull request:
Hello,

This PR fixes the "file exists" error when the remote MSSQL server is not configured in english (see https://github.com/fortra/impacket/issues/1998).

Now it works no matter what the language is:

<img width="860" height="618" alt="image" src="https://github.com/user-attachments/assets/39f181e1-d4bc-4404-8673-5514289d16e5" />

:)

(This PR requires https://github.com/fortra/impacket/pull/2001 to be merged first)

gabrielg5 commented:
Hey @Dfte, hello!

Been checking this one. Some notes:
* We should add the `impacket.mssql` package in `setup.py`
* In the `version_name` property of `MSSQL_VERSION` class, what if we just _pass_ on `KeyError`?
I tried with a couple SQLServer instances and got `Microsoft SQL Server 2014 (unknown) (12.0.6179)` and `Microsoft SQL Server 2019 (unknown) (15.0.2135)` for example. If we just _pass_ on that exception at worst we'd say `Microsoft SQL Server (<major>.<minor>.<build>)`

As an FYI only because change is ok :)... there's a boolean parameter `tuplemode` in the `sql_query` function to return rows as list instead than dictionary. Perhaps useful in the scenario of the _"File Exists"_ fix

(Think I'll be only checking this one and close #2001 as those changes are also included in here)

Dfte commented:
Ayo!! I have added impacket.mssql to the setupd.py and modified the version_name to pass on keyerror exception.

As for the file exist dict/list, good to know but the patch I have wrote should be enough :)!

Let me know if you need anything else :)

gabrielg5 commented:
Thank you!
merging now!!
